src/formats/bbcode.js

```diff
@@ -50,6 +50,21 @@
 		auto: 3
 	};
 
+	var MatchingMode = {
```

Comment:
I'm wondering about this, do you think there will ever be a need for anything other than `all` and `any` matching? I'm thinking anything else is likely to be very complex and require a different API.

If not, using a flag like `strictMatching: true` (not sure about the name) might be simpler. Or if you think there might be more matching types then using strings might be better.

I know `quoteType` does it like this but I'm not a massive fan of that either. It was an attempt to improve autocompletion but it didn't really help and ended up not being very JS like. I've been wanting to replace it with plain strings at some point as, if typescript definitions are added, the autocompletion will work just as well if not better and would be more JS like. What do you think?

Reply:
My view on the enums is that they help make the public API easier to use and understand. That said, a simple bool here could suffice given a descriptive name. Maybe `matchAllAttrs`. I still think that it may not be as meaningful as a full enum. I'm kinda torn on this.

Reply:
I just thought of a possibility to add two new possible types
-   `ArrayList` or a`SimpleArray`
- `Callback`

Reply:
What would ArrayList / SimpleArray do? Callback could be useful although that can be done inside of `format` and if not, just setting a function and having `handleTags` check for it would likely be safer than relying on matching mode being set..

If there can be multiple modes, just using strings would probably be more JS like, it's what I want to do with `quoteType` as that ended up being clunky to type out without any real benefits.

Reply:
I'll make this a string then as it seems that strings are used everywhere like you said. Enums seem to be an anomaly in JS (but are common in compiled languages such as C#)

`array` would just be an array like that in the `styles` attributive and would match with `Array. prototype.includes()`.

Reply:
Ehh, maybe not. Magic string seem too highly error prone due to typos.